azavea/cac-tripplanner
# Display events

## Overview

- Display events in the UI
- Incorporate styling changes from #960 and implement new elements
- Assorted related improvements and refactoring related to place list display

### Demo

Event cards on home page:
![image](https://user-images.githubusercontent.com/960264/34593576-ad32353a-f199-11e7-904b-07c4fd6285d8.png)

Head of detail page for single-day event with a destination:
![image](https://user-images.githubusercontent.com/960264/34593625-0e5b6eda-f19a-11e7-8483-54edd9b504c6.png)

Detail page of event without destination and different start and end dates:
![image](https://user-images.githubusercontent.com/960264/34593639-2e1cae46-f19a-11e7-9353-77767ddfaaf3.png)

Filtering for events by isochrone:
![image](https://user-images.githubusercontent.com/960264/34593672-5d5d16d2-f19a-11e7-8443-b1047a0d4b55.png)

Filter message on no events found:
![image](https://user-images.githubusercontent.com/960264/34593717-9636050e-f19a-11e7-8642-3edeebf7ac65.png)


### Notes

@lederer: the styling for event times assumed start and end day would be the same. Modified line height to accommodate two lines of text for start and end date and time when they differ. Might want to revisit how that is displayed.

Events have an optional destination, which gives the event its location, if it has any. When places are sorted by travel time, events without a destination sort last. When no origin set, events display below destinations under the 'All' filter.


## Testing Instructions

 * `vagrant up`
 * `vagrant ssh app`
 * Run migrations: `cd /opt/app/python/cac_tripplanner && ./manage.py migrate`
 * Rebuild JS: `cd /opt/app/src && npm run gulp-development`
 * PR bumps service worker version number, so cache clear should *not* be needed
 * Add at least two events: http://localhost:8024/admin/destinations/event/add/
 * At least one event should have an associated destination; at least one should have none set
 * Events should display as expected in user-facing app
 * Filtering and "explore" mode travelshed filtering should function as expected
 * Events with a destination set should show in the autocomplete results (those without should not)


## Checklist

Closes #911 
Closes #935 
Closes #906 
Closes #945
Fixes #953 
Fixes #964
